no element named `get_io_service`

When installing the package with pacman the following errors appear in my up2date arch environment.

```
Scanning dependencies of target mavconn
[ 16%] Building CXX object CMakeFiles/mavconn.dir/src/tcp.cpp.o
[ 66%] Building CXX object CMakeFiles/mavconn.dir/src/mavlink_helpers.cpp.o
[ 66%] Building CXX object CMakeFiles/mavconn.dir/src/serial.cpp.o
[ 66%] Building CXX object CMakeFiles/mavconn.dir/src/interface.cpp.o
[ 83%] Building CXX object CMakeFiles/mavconn.dir/src/udp.cpp.o
/tmp/yaourt-tmp-korny/aur-ros-melodic-libmavconn/src/mavros-release-release-melodic-libmavconn-0.26.3-0/src/tcp.cpp: In Elementfunktion »void mavconn::MAVConnTCPClient::client_connected(size_t)«:
/tmp/yaourt-tmp-korny/aur-ros-melodic-libmavconn/src/mavros-release-release-melodic-libmavconn-0.26.3-0/src/tcp.cpp:123:9: Fehler: »boost::asio::ip::tcp::socket« {aka »class boost::asio::basic_stream_socket<boost::asio::ip::tcp>«} hat kein Element namens »get_io_service«
  123 |  socket.get_io_service().post(std::bind(&MAVConnTCPClient::do_recv, shared_from_this()));
      |         ^~~~~~~~~~~~~~
/tmp/yaourt-tmp-korny/aur-ros-melodic-libmavconn/src/mavros-release-release-melodic-libmavconn-0.26.3-0/src/tcp.cpp: In Elementfunktion »virtual void mavconn::MAVConnTCPClient::send_bytes(const uint8_t*, size_t)«:
/tmp/yaourt-tmp-korny/aur-ros-melodic-libmavconn/src/mavros-release-release-melodic-libmavconn-0.26.3-0/src/tcp.cpp:169:9: Fehler: »boost::asio::ip::tcp::socket« {aka »class boost::asio::basic_stream_socket<boost::asio::ip::tcp>«} hat kein Element namens »get_io_service«
  169 |  socket.get_io_service().post(std::bind(&MAVConnTCPClient::do_send, shared_from_this(), true));
      |         ^~~~~~~~~~~~~~
/tmp/yaourt-tmp-korny/aur-ros-melodic-libmavconn/src/mavros-release-release-melodic-libmavconn-0.26.3-0/src/tcp.cpp: In Elementfunktion »virtual void mavconn::MAVConnTCPClient::send_message(const mavlink_message_t*)«:
/tmp/yaourt-tmp-korny/aur-ros-melodic-libmavconn/src/mavros-release-release-melodic-libmavconn-0.26.3-0/src/tcp.cpp:191:9: Fehler: »boost::asio::ip::tcp::socket« {aka »class boost::asio::basic_stream_socket<boost::asio::ip::tcp>«} hat kein Element namens »get_io_service«
  191 |  socket.get_io_service().post(std::bind(&MAVConnTCPClient::do_send, shared_from_this(), true));
      |         ^~~~~~~~~~~~~~
/tmp/yaourt-tmp-korny/aur-ros-melodic-libmavconn/src/mavros-release-release-melodic-libmavconn-0.26.3-0/src/tcp.cpp: In Elementfunktion »virtual void mavconn::MAVConnTCPClient::send_message(const mavlink::Message&)«:
/tmp/yaourt-tmp-korny/aur-ros-melodic-libmavconn/src/mavros-release-release-melodic-libmavconn-0.26.3-0/src/tcp.cpp:211:9: Fehler: »boost::asio::ip::tcp::socket« {aka »class boost::asio::basic_stream_socket<boost::asio::ip::tcp>«} hat kein Element namens »get_io_service«
  211 |  socket.get_io_service().post(std::bind(&MAVConnTCPClient::do_send, shared_from_this(), true));
      |         ^~~~~~~~~~~~~~
make[2]: *** [CMakeFiles/mavconn.dir/build.make:115: CMakeFiles/mavconn.dir/src/tcp.cpp.o] Fehler 1
make[2]: *** Es wird auf noch nicht beendete Prozesse gewartet....
make[1]: *** [CMakeFiles/Makefile2:416: CMakeFiles/mavconn.dir/all] Fehler 2
make: *** [Makefile:130: all] Fehler 2
==> FEHLER: Ein Fehler geschah in build().
    Breche ab...
==> FEHLER:Makepkg konnte ros-melodic-libmavconn nicht erstellen.

```

--- a/.SRCINFO
+++ b/.SRCINFO
@@ -1,8 +1,8 @@
 pkgbase = ros-melodic-libmavconn
 	pkgdesc = ROS - MAVLink communication library.
-	pkgver = 0.32.2
-	pkgrel = 1
-	url = https://wiki.ros.org/mavros
+	pkgver = 0.33.3
+	pkgrel = 2
+	url = https://wiki.ros.org/libmavconn
 	arch = any
 	license = GPLv3, LGPLv3, BSD
 	makedepends = cmake
@@ -14,8 +14,8 @@ pkgbase = ros-melodic-libmavconn
 	depends = ros-melodic-mavlink
 	depends = boost
 	depends = console-bridge
-	source = ros-melodic-libmavconn-0.32.2.tar.gz::https://github.com/mavlink/mavros/archive/0.32.2.tar.gz
-	sha256sums = d5e04661f88896a9a77f22a083023ce5e8e13dd71c1f8f1b7d5e187b3bf6c4f8
+	source = ros-melodic-libmavconn-0.33.3.tar.gz::https://github.com/mavlink/mavros/archive/0.33.3.tar.gz
+	sha256sums = eb4fc2439c78cdc2fa5f2d9ab81abe4f1fa54f9dd45f02b8f2440a231125118e
 
 pkgname = ros-melodic-libmavconn
 

--- a/.SRCINFO
+++ b/.SRCINFO
@@ -1,7 +1,7 @@
 pkgbase = ros-melodic-libmavconn
 	pkgdesc = ROS - MAVLink communication library.
 	pkgver = 0.33.3
-	pkgrel = 2
+	pkgrel = 1
 	url = https://wiki.ros.org/libmavconn
 	arch = any
 	license = GPLv3, LGPLv3, BSD

--- a/.SRCINFO
+++ b/.SRCINFO
@@ -15,7 +15,9 @@ pkgbase = ros-melodic-libmavconn
 	depends = boost
 	depends = console-bridge
 	source = ros-melodic-libmavconn-0.33.3.tar.gz::https://github.com/mavlink/mavros/archive/0.33.3.tar.gz
+	source = boost-1.70.patch
 	sha256sums = eb4fc2439c78cdc2fa5f2d9ab81abe4f1fa54f9dd45f02b8f2440a231125118e
+	sha256sums = d7a6c462f9328536ffd9b5b59e3a7e483fc5d403d71114bfc03024c364df233d
 
 pkgname = ros-melodic-libmavconn
 

--- a/PKGBUILD
+++ b/PKGBUILD
@@ -1,12 +1,12 @@
 # Script generated with import_catkin_packages.py
 # For more information: https://github.com/bchretien/arch-ros-stacks
 pkgdesc="ROS - MAVLink communication library."
-url='https://wiki.ros.org/mavros'
+url='https://wiki.ros.org/libmavconn'
 
 pkgname='ros-melodic-libmavconn'
-pkgver='0.32.2'
+pkgver='0.33.3'
 arch=('any')
-pkgrel=1
+pkgrel=2
 license=('GPLv3, LGPLv3, BSD')
 
 ros_makedepends=(ros-melodic-mavlink
@@ -21,16 +21,9 @@ depends=(${ros_depends[@]}
   boost
   console-bridge)
 
-# Git version (e.g. for debugging)
-# _tag=release/melodic/libmavconn/${pkgver}-${_pkgver_patch}
-# _dir=${pkgname}
-# source=("${_dir}"::"git+https://github.com/mavlink/mavros-release.git"#tag=${_tag})
-# sha256sums=('SKIP')
-
-# Tarball version (faster download)
 _dir="mavros-${pkgver}/libmavconn"
 source=("${pkgname}-${pkgver}.tar.gz"::"https://github.com/mavlink/mavros/archive/${pkgver}.tar.gz")
-sha256sums=('d5e04661f88896a9a77f22a083023ce5e8e13dd71c1f8f1b7d5e187b3bf6c4f8')
+sha256sums=('eb4fc2439c78cdc2fa5f2d9ab81abe4f1fa54f9dd45f02b8f2440a231125118e')
 
 build() {
   # Use ROS environment variables

--- a/PKGBUILD
+++ b/PKGBUILD
@@ -6,7 +6,7 @@ url='https://wiki.ros.org/libmavconn'
 pkgname='ros-melodic-libmavconn'
 pkgver='0.33.3'
 arch=('any')
-pkgrel=2
+pkgrel=1
 license=('GPLv3, LGPLv3, BSD')
 
 ros_makedepends=(ros-melodic-mavlink

--- a/PKGBUILD
+++ b/PKGBUILD
@@ -22,8 +22,15 @@ depends=(${ros_depends[@]}
   console-bridge)
 
 _dir="mavros-${pkgver}/libmavconn"
-source=("${pkgname}-${pkgver}.tar.gz"::"https://github.com/mavlink/mavros/archive/${pkgver}.tar.gz")
-sha256sums=('eb4fc2439c78cdc2fa5f2d9ab81abe4f1fa54f9dd45f02b8f2440a231125118e')
+source=("${pkgname}-${pkgver}.tar.gz"::"https://github.com/mavlink/mavros/archive/${pkgver}.tar.gz"
+        "boost-1.70.patch")
+sha256sums=('eb4fc2439c78cdc2fa5f2d9ab81abe4f1fa54f9dd45f02b8f2440a231125118e'
+            'd7a6c462f9328536ffd9b5b59e3a7e483fc5d403d71114bfc03024c364df233d')
+
+prepare() {
+    cd "${srcdir}/${_dir}"
+    patch -uN src/tcp.cpp ../../../boost-1.70.patch || return 1
+}
 
 build() {
   # Use ROS environment variables

--- a/boost-1.70.patch
+++ b/boost-1.70.patch
@@ -1,0 +1,52 @@
+--- tcp.cpp.old	2019-11-16 23:18:03.959450362 -0500
++++ tcp.cpp	2019-11-16 23:14:48.329447262 -0500
+@@ -21,6 +21,13 @@
+ #include <mavconn/thread_utils.h>
+ #include <mavconn/tcp.h>
+ 
++// Ensure the correct io_service() is called based on boost version
++#if BOOST_VERSION >= 107000
++#define GET_IO_SERVICE(s) ((boost::asio::io_context&)(s).get_executor().context())
++#else
++#define GET_IO_SERVICE(s) ((s).get_io_service())
++#endif
++
+ namespace mavconn {
+ 
+ using boost::system::error_code;
+@@ -120,7 +127,7 @@
+ 			server_channel, conn_id, to_string_ss(server_ep).c_str());
+ 
+ 	// start recv
+-	socket.get_io_service().post(std::bind(&MAVConnTCPClient::do_recv, shared_from_this()));
++	GET_IO_SERVICE(socket).post(std::bind(&MAVConnTCPClient::do_recv, shared_from_this()));
+ }
+ 
+ MAVConnTCPClient::~MAVConnTCPClient()
+@@ -166,7 +173,7 @@
+ 
+ 		tx_q.emplace_back(bytes, length);
+ 	}
+-	socket.get_io_service().post(std::bind(&MAVConnTCPClient::do_send, shared_from_this(), true));
++	GET_IO_SERVICE(socket).post(std::bind(&MAVConnTCPClient::do_send, shared_from_this(), true));
+ }
+ 
+ void MAVConnTCPClient::send_message(const mavlink_message_t *message)
+@@ -188,7 +195,7 @@
+ 
+ 		tx_q.emplace_back(message);
+ 	}
+-	socket.get_io_service().post(std::bind(&MAVConnTCPClient::do_send, shared_from_this(), true));
++	GET_IO_SERVICE(socket).post(std::bind(&MAVConnTCPClient::do_send, shared_from_this(), true));
+ }
+ 
+ void MAVConnTCPClient::send_message(const mavlink::Message &message, const uint8_t source_compid)
+@@ -208,7 +215,7 @@
+ 
+ 		tx_q.emplace_back(message, get_status_p(), sys_id, source_compid);
+ 	}
+-	socket.get_io_service().post(std::bind(&MAVConnTCPClient::do_send, shared_from_this(), true));
++	GET_IO_SERVICE(socket).post(std::bind(&MAVConnTCPClient::do_send, shared_from_this(), true));
+ }
+ 
+ void MAVConnTCPClient::do_recv()

